ibtk/include/ibtk/ibtk_utilities.h

```diff
@@ -167,35 +167,35 @@ level_can_be_refined(int level_number, int max_levels)
     return level_number < finest_level_number;
 }
 
-typedef Eigen::Matrix<double, 2, 2> Matrix2d;
-typedef Eigen::Matrix<double, 2, 1> Vector2d;
-typedef Eigen::Matrix<double, 2, 1> ColumnVector2d;
-typedef Eigen::Matrix<double, 1, 2> RowVector2d;
-
-typedef Eigen::Matrix<double, 3, 3> Matrix3d;
-typedef Eigen::Matrix<double, 3, 1> Vector3d;
-typedef Eigen::Matrix<double, 3, 1> ColumnVector3d;
-typedef Eigen::Matrix<double, 1, 3> RowVector3d;
-
-typedef Eigen::Matrix<double, NDIM, NDIM> MatrixNd;
-typedef Eigen::Matrix<double, NDIM, 1> VectorNd;
-typedef Eigen::Matrix<double, NDIM, 1> ColumnVectorNd;
-typedef Eigen::Matrix<double, 1, NDIM> RowVectorNd;
-
-typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> MatrixXd;
-typedef Eigen::Matrix<double, Eigen::Dynamic, 1> VectorXd;
-typedef Eigen::Matrix<double, Eigen::Dynamic, 1> ColumnVectorXd;
-typedef Eigen::Matrix<double, 1, Eigen::Dynamic> RowVectorXd;
-
-typedef MatrixNd Matrix;
-typedef VectorNd Point;
-typedef VectorNd Vector;
+using Matrix2d = Eigen::Matrix<double, 2, 2>;
+using Vector2d = Eigen::Matrix<double, 2, 1>;
+using ColumnVector2d = Eigen::Matrix<double, 2, 1>;
+using RowVector2d = Eigen::Matrix<double, 1, 2>;
+
+using Matrix3d = Eigen::Matrix<double, 3, 3>;
+using Vector3d = Eigen::Matrix<double, 3, 1>;
+using ColumnVector3d = Eigen::Matrix<double, 3, 1>;
+using RowVector3d = Eigen::Matrix<double, 1, 3>;
+
+using MatrixNd = Eigen::Matrix<double, NDIM, NDIM>;
+using VectorNd = Eigen::Matrix<double, NDIM, 1>;
+using ColumnVectorNd = Eigen::Matrix<double, NDIM, 1>;
+using RowVectorNd = Eigen::Matrix<double, 1, NDIM>;
+
+using MatrixXd = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>;
+using VectorXd = Eigen::Matrix<double, Eigen::Dynamic, 1>;
+using ColumnVectorXd = Eigen::Matrix<double, Eigen::Dynamic, 1>;
+using RowVectorXd = Eigen::Matrix<double, 1, Eigen::Dynamic>;
```

Comment:
@boyceg Eigen already has most of these typedefs, e.g.
`typedef Eigen::Matrix<double, 3, 3> Eigen::Matrix3d;`
Would it be worth it to get rid of these in favor of Eigen's typedefs?

Reply:
As far as I can tell, the only place these typedefs are used are in `GeneralizedIBMethod.cpp` and `IBKirchoffRodForceGen.cpp`. All other uses of Matrix/Vector2/3d utilize the Eigen typedef.

Reply:
Hmmm, why did I add in those `typedef`s?

The reason is that I used to do this stuff with Blitz++ classes, but then Blitz++ development slowed down, and I switched to Eigen. I put in the typedefs to make the IBAMR/IBTK interface more agnostic to the linear algebra backend.

I am sure that @drwells will chime in about whether this makes sense or not. :-)

Reply:
(Also it is certainly possible that I put the `typedef`s with the _intention_ of "hiding" the backend but then forgot to follow through on it...)

Reply:
Yes we could go the other way and replace all uses of Eigen's `typedef`s with these.

Reply:
Let's see what @drwells says...

Reply:
I don't think we should worry about being agnostic w.r.t. the dense linear algebra background.

It looks like `Eigen` already offers `Eigen::VectorXd` (and we use it in `IBFEPatchRecoveryPostProcessor.cpp`). Since there is some overlap between what we use as `Eigen::VectorXd` and `IBTK::Vector` I think it would be better to just stick to the eigen-provided using declarations.

Either way: we can fix this in a follow-up PR since we do use `IBTK::Vector` a lot. I suggest we use `[[deprecated]]` or similar to mark these as deprecated and switch all of our stuff to use, e.g., `Eigen::VectorXd`.